### DISCUSSION TITLE
Is API usage possible?

### DISCUSSION THREAD

**Discussion by jalfd** [Mogztter/asciidoctor-web-pdf#712] [Q&A] [answered]
All the examples in the README file show invoking asciidoctor-web-pdf as a CLI tool.

Is that the only intended usage, or is it also possible to use this as an API?

Currently I have a script wich calls `asciidoctor.convert()` to generate HTML, but I'd like to do the same with asciidoctor-web-pdf to generate a PDF version.
Is there a well-supported way to do this?

**Comment by ggrossetie** [comment, marked as answer]
Hey! 
Asciidoctor Web PDF relies on an headless browser (through Puppeteer). You can invoke Asciidoctor Web PDF programmatically but it will start an headless browser that's the main reason why we didn't introduce a proper API.
